core/network/ipaddress_test.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package network_test
+
+import (
+	"net/netip"
+
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/core/network"
+	"github.com/juju/juju/internal/testing"
+)
+
+type IPAddressSuite struct {
+	testing.BaseSuite
+}
+
+var _ = gc.Suite(&IPAddressSuite{})
+
+func (s *IPAddressSuite) TestIPv4AddressSplit(c *gc.C) {
+	addr := network.IPAddress{netip.MustParseAddr("192.168.0.0")}
+	msb, lsb := addr.AsInts()
+	c.Assert(msb, gc.Equals, uint64(0))
+	c.Assert(lsb, gc.Equals, uint64(0xc0a80000))
+}
+
+func (s *IPAddressSuite) TestIPv6AddressSplit(c *gc.C) {
+	addr := network.IPAddress{netip.MustParseAddr("fd7a:115c:a1e0:ab12:4843:cd96:626b:430b")}
```

Comment:
Small nit but can you use the `2001:0DB8::/32` documentation range in tests ?

Reply:
Sorry what I mean is `2001:0DB8::/32` is defined in an RFC as a reserved documentation range that should be used when testing and documenting ipv6. It was made to avoid the case with ipv4 where everyone typed 1.1.1.1 in their code as a testing address.

Just asking to pick a random address in that range so we follow the recommendations in the RFC.